fix(spec,tests): Change BAL to List[AccountChange]

gurukamath opened the pull request:
## 🗒️ Description

The EIP does the follwoing updates

1. Defines BlockAccessList = List[AccountChanges]. The specs currently have block access list as a dataclass which has an attribute calles account changes which is this list. This inconsistency creates issue when we try to decode amsterdam block rlp to block.
2. The EIP also consumes the newly created amsterdam json fixtures in json_infra

## ✅ Checklist
<!-- Please check off all required items. For those that don't apply remove them accordingly. -->

- [x] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx tox -e static
    ```
- [x] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.

#### Cute Animal Picture

![Cute Animals - 12 of 12](https://github.com/user-attachments/assets/9733552a-e7fe-49ef-a6eb-884d2963bcf1)

gurukamath committed:
fix(spec): update BAL type

gurukamath committed:
fix(tests): run amsterdam jsons

fselmo committed:
fix(test-specs): Add BAL to genesis beyond Amsterdam fork

fselmo reviewed (approved):
@gurukamath I added the bal in the genesis here and fixed the fill with your changes. Lmk what you think but this lgtm.

gurukamath commented:
> @gurukamath I added the bal in the genesis here and fixed the fill with your changes. Lmk what you think but this lgtm.

LGTM! However, the genesis RLP check is enabled now for `json_infra` but the fixtures don't yet have the changes you made. So the `json_infra` blockchain tests will continue to fail until we upgrade to a newer release which has the genesis fix. I'm ok with merging this in and then updating to a new release in another PR

fselmo commented:
> I'm ok with merging this in and then updating to a new release in another PR

Right, this was my expectation here. We plan to make a new release this week 👍🏼
